### ISSUE TITLE
Add identifier types to AbstractSpecies base class

### ISSUE THREAD

**Issue by JR-1991** [EnzymeML/PyEnzyme#56]
**Description**

Proteins and reactants may be purchased at vendors and thus possess serial numbers or DOIs which lead to more information. Thus, it is necessary to mirror this in EnzymeML.

**Solution**

Since identifiers appear in many flavors, the EnzymeML data model needs to accommodate for a variety of identifier types. Hence, it may be necessary to define a class that holds the actual persistent identifier and its scheme. This would result in the following XML:

```xml
<annotation>
	<enzymeml:persistentIdentifiers>
		<enzymeml:pID value="10.1093/ajae/aaq063" type="DOI" />
		<enzymeml:pID value="873673647367436" type="Serialnumber" />
	</enzymeml:persistentIdentifiers>
</annotation>
```

**Comment by fbergmann** [comment]
isnt serial number to generic? You would probably have to know specifics, to be able to resolve it later on. You might need more than one field to describe that. Maybe adding to the identifiers.org registry would be a better idea, from there you'd just need to store urls. Ensuring they could be resolved later: 

doi example <https://registry.identifiers.org/registry/doi>

**Comment by JR-1991** [comment]
> isnt serial number to generic? You would probably have to know specifics, to be able to resolve it later on. You might need more than one field to describe that. Maybe adding to the identifiers.org registry would be a better idea, from there you'd just need to store urls. Ensuring they could be resolved later:
> 
> doi example https://registry.identifiers.org/registry/doi

Makes sense, so instead of the plain serial number/identifier we'd provide a URL to identifiers.org, given a schema exists? The only concern I have is that this requires these schemes to be registered beforehand.

As an intermediate solution in the case of a product, would it make sense to store the URL to the product page as well as the identifier? The URL might change within years, but the ID and the hint to the provider given in the URL would persist.

**Comment by itbjpl** [comment]
We would refer to a DOI, if the synthesis of the catalyst or the compound is described elsewhere in a paper or in a dataset. This would work well via the identifiers.org registry  
But how can we refer to an enzyme or a compound, which has been supplied by a commercial vendor? We should even identify the lot or batch.

**Comment by fbergmann** [comment]
> But how can we refer to an enzyme or a compound, which has been supplied by a commercial vendor? We should even identify the lot or batch.

this is exactly what i meant, by just referencing a generic serial number. it would be hard to be able to resolve it back, there could be an arbitrary number of fields needed to describe it. Of course it would always be possible to encode these parameters in an url. If it is not one of identifiers.org, as in the well known example (say for ec / kegg compounds) it could be something like:

```
https://enzymeml.org/compound?company=<companyname>&serialNo=<number>&batch.=...... 
```

or some such.

**Comment by JR-1991** [comment]
Sounds like a plan! According to your example the URL would include the following parameters:

* company
* serialNo
* batch

@itbjpl anything other we should add? Once we compiled everything I'd register it.

**Comment by fbergmann** [comment]
that was just an example to point out that this way arbitrary parameters could be encoded that another tool could then later interpret. This will be needed, since a serial number alone is often not enough information to get back to what it is.  I'm not suggesting to include those 3 specifically.

**Comment by JR-1991** [comment]
Of course, it's a good starting point though in my opinion. We could work on a concrete schema within this issue or the webinar?
